src/main.rs

```diff
@@ -157,19 +157,25 @@ fn handle_generate(generate: &Generate) -> Result<()> {
 }
 
 fn handle_attenuate(attenuate: &Attenuate) -> Result<()> {
-    let biscuit_format = if attenuate.raw_input {
+    let biscuit_format = if attenuate.biscuit_input_args.raw_input {
         BiscuitFormat::RawBiscuit
     } else {
         BiscuitFormat::Base64Biscuit
     };
 
-    let biscuit_from = if attenuate.biscuit_file == PathBuf::from("-") {
+    let biscuit_from = if attenuate.biscuit_input_args.biscuit_file == PathBuf::from("-") {
         BiscuitBytes::FromStdin(biscuit_format)
     } else {
-        BiscuitBytes::FromFile(biscuit_format, attenuate.biscuit_file.clone())
```

Comment:
`.clone()` duplicate `biscuit_file` "object" in order to allow a potential reuse of value in the next lines.
By default, `BiscuitBytes::FromFile` will capture the ownership of `biscuit_file` and this will be no more accessible.

Reply:
But as I understand, clone() copies the file, is it effective? Why don't just use the reference to the file?

Reply:
Nop, .clone() copy the value of `biscuit_file` which is a `PathBuf` structure (that just store the file path).
`biscuit_file` is a field of `Attenuate` struct, which is (re)used later in the code (line 176).
You can't move the ownership of just a field of a struct.

Reply:
Still don't get the idea, but OK :)

Reply:
You can read the [ownership introduction](https://doc.rust-lang.org/book/ch04-00-understanding-ownership.html) in the Rust book, this is clearly the most destabilizing aspect of Rust, but once integrated, everything seems logical.